iree/tools/translate_main.cc

```diff
@@ -108,8 +109,8 @@ int main(int argc, char **argv) {
   /// Processes the memory buffer with a new MLIRContext.
   auto processBuffer = [&](std::unique_ptr<llvm::MemoryBuffer> ownedBuffer,
                            llvm::raw_ostream &os) {
-    mlir::MLIRContext context;
-    context.loadAllGloballyRegisteredDialects();
+    mlir::MLIRContext context(/*loadAllDialects=*/false);
+    registry.appendTo(context.getDialectRegistry());
     llvm::SourceMgr sourceMgr;
     sourceMgr.AddNewSourceBuffer(std::move(ownedBuffer), llvm::SMLoc());
```

Comment:
Same as for the opt tool, upstream has a more convenient entry point: https://github.com/llvm/llvm-project/blob/master/mlir/tools/mlir-translate/mlir-translate.cpp#L35

Reply:
Added a TODO